lib/util/LabelUtil.js

```diff
@@ -251,7 +251,10 @@ export function setLabel(element, text) {
   if (attr) {
 
     if (attr === 'categoryValueRef') {
-      semantic['categoryValueRef'].value = text;
+      if (!semantic[attr]) {
```

Comment:
```suggestion
      
      // ignore update to non-existing category value when removing the text
      if (!text && !semantic[attr]) {
```

Reply:
This would match my first preferred suggestion of my initial approach: https://github.com/bpmn-io/bpmn-js/issues/2231#issuecomment-2351201757

It should be safe for all use cases - at least those I came across.